### PR TITLE
Temporary fix to not download Istio WASM filters until we can fix the integration tests

### PR DESCRIPTION
**Pull request by brian-avery**
Currently in our integration tests, we download several dependencies from Istio. This fixes the build for WASM by making it so that it doesn't try to download the Istio dependencies. Note that this doesn't fix the integration tests, it just bypasses them.

### CODE REVIEW

**Review by jwendell — CHANGES_REQUESTED** [review]
this doesn't look right

**Review by jwendell — CHANGES_REQUESTED** [review]
you're missing the patch file. Also, improve the title

File: istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
++#for plugin in stats metadata_exchange
++#do
++#  FILTER_WASM_URL="${ISTIO_ENVOY_BASE_URL}/${plugin}-${ISTIO_ENVOY_VERSION}.wasm"
++#  download_wasm_if_necessary "${FILTER_WASM_URL}" "${WASM_RELEASE_DIR}"/"${plugin//_/-}"-filter.wasm
++#done
+ 
+ # Copy native envoy binary to ISTIO_OUT
+ echo "Copying ${ISTIO_ENVOY_NATIVE_PATH} to ${ISTIO_OUT}/envoy"
+diff --git a/pilot/docker/Dockerfile.proxyv2 b/pilot/docker/Dockerfile.proxyv2
+index b459df9b3..c6ee9ae70 100644
```
Comment (jwendell) [inline istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch:38]: The code from here to below never runs, so, they are not necessary, right?

File: istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ fi
+ 
+-# Donwload WebAssembly plugin files
+-WASM_RELEASE_DIR=${ISTIO_ENVOY_LINUX_RELEASE_DIR}
+-for plugin in stats metadata_exchange
+-do
+-  FILTER_WASM_URL="${ISTIO_ENVOY_BASE_URL}/${plugin}-${ISTIO_ENVOY_VERSION}.wasm"
+-  download_wasm_if_necessary "${FILTER_WASM_URL}" "${WASM_RELEASE_DIR}"/"${plugin//_/-}"-filter.wasm
+-done
++# Because we separate Istio and proxy builds, we don't need to have these files
```
Comment (jwendell) [inline istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch:27]: Instead of commenting, you could just delete this block.

**Review by jwendell — CHANGES_REQUESTED** [review]

File: istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch (hunk 2 above)
Comment (brian-avery) [reply istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch:27]: This is all temporary fixes until we can fix the integration tests. So I'd like to merge as is if it's good.

**Review by bison — APPROVED** [review]
The Docker related make targets are super confusing to me, but if this gets the build working, then LGTM.

File: istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch (hunk 1 above)
Comment (brian-avery) [reply istio/0001-Disable-wasm-download-until-we-re-ready-for-it.patch:38]: Some of the code below does run and fails. The dockerfile changes themselves are likely not needed, but again, this is all temporary fixes until we can make the integration tests work.

**Review by jwendell — APPROVED** [review]

**Pull request merged**